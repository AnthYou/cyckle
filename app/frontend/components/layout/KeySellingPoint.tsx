import React from "react";

interface KeySellingPointProps {
  id: number;
  title: string;
  content: string;
  image: string;
  alt: string
}

const KeySellingPoint = ({
  id,
  title,
  content,
  image,
  alt,
}: KeySellingPointProps): JSX.Element => {
  return (
    <div>
      <div>
        <h2>{title}</h2>
        <p>{content}</p>
      </div>
      <img src={image} alt={alt} />
    </div>
  );
};

export default KeySellingPoint;
