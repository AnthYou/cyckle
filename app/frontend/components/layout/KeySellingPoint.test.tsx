import React from "react";
import renderer from "react-test-renderer";

import KeySellingPoint from "./KeySellingPoint";
import Blob from "@/images/blob.svg";

describe("<KeySellingPoint />", () => {
  const defaultProps = {
    id: 1,
    title: "Test",
    content: "Lorem ipsum",
    image: Blob,
    alt: "This is a blob."
  };

  const wrapper = renderer.create(<KeySellingPoint {...defaultProps} />);

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
