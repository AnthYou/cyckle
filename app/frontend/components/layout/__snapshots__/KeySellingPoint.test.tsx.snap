// Vitest Snapshot v1

exports[`<KeySellingPoint /> > should render correctly 1`] = `
<div>
  <div>
    <h2>
      Test
    </h2>
    <p>
      Lorem ipsum
    </p>
  </div>
  <img
    alt="This is a blob."
    src="/@fs/@/images/blob.svg"
  />
</div>
`;
